domain/removal/state/state.go

```diff
@@ -22,3 +28,87 @@ func NewState(factory database.TxnRunnerFactory, logger logger.Logger) *State {
 		logger:    logger,
 	}
 }
+
+// RelationExists returns true if a relation exists with the input UUID.
+func (st *State) RelationExists(ctx context.Context, rUUID string) (bool, error) {
+	db, err := st.DB()
+	if err != nil {
+		return false, errors.Capture(err)
+	}
+
+	relationUUID := entityUUID{UUID: rUUID}
+	existsStmt, err := st.Prepare(`
+SELECT uuid AS &entityUUID.uuid
+FROM   relation
+WHERE  uuid = $entityUUID.uuid`, relationUUID)
+	if err != nil {
+		return false, errors.Errorf("preparing relation exists query: %w", err)
+	}
+
+	var relationExists bool
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, existsStmt, relationUUID).Get(&relationUUID)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return nil
+		} else if err != nil {
+			return errors.Errorf("running relation exists query: %w", err)
+		}
+
+		relationExists = true
+		return nil
+	})
+
+	return relationExists, err
+}
+
+// RelationAdvanceLifeAndScheduleRemoval advances the life cycle of the relation
+// with the input UUID to dying if it is alive, and schedules a removal job for
+// the relation, qualified with the input force boolean.
+// We don't care if the relation does not exist at this point because:
+// - it should have been validated prior to calling this method,
+// - the removal job executor will handle that fact.
+func (st *State) RelationAdvanceLifeAndScheduleRemoval(
+	ctx context.Context, removalUUID, relUUID string, force bool, when time.Time,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	relationUUID := entityUUID{UUID: relUUID}
+	lifeStmt, err := st.Prepare(`
+UPDATE relation
+SET    life_id = 1
+WHERE  uuid = $entityUUID.uuid
+AND    life_id = 0`, relationUUID)
+	if err != nil {
+		return errors.Errorf("preparing relation life update: %w", err)
+	}
+
+	removalRec := removal{
+		UUID:          removalUUID,
+		RemovalTypeID: 0,
```

Comment:
todo: get the `RemovalTypeID` from the `removal_type` table rather than hard coding it.

Reply:
This is the common pattern, just have a test the ensures that both are the same.

Reply:
This part will be DRY'd up as we add more job types anyway. I've changed the test to join on the `removal_type` table and ensure that the job type is _relation_.